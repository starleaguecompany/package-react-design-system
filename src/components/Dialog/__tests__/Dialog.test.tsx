import * as React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Dialog, Icon, Button } from '../../../index';
import { Menu } from '@starleaguecompany/react-icons';

describe('Dialog', () => {
  test('should render correctly with no props', async () => {
    const { container } = render(<Dialog />);

    expect(container).toMatchSnapshot();
    expect(screen.queryByTestId('Dialog')).toBeNull();
  });

  test('should render correctly with visible prop', async () => {
    render(<Dialog visible />);

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    await waitFor(() => expect(screen.getByTestId('Dialog')).toBeVisible());
  });

  test('should render correctly with attributes', () => {
    render(<Dialog id="test-id" className="test-class" visible />);

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    expect(screen.getByTestId('Dialog')).toHaveClass('test-class');
    expect(screen.getByTestId('Dialog')).toHaveAttribute('id', 'test-id');
  });

  test('should render correctly with content', async () => {
    render(
      <Dialog visible>
        <Dialog.Content>Test content</Dialog.Content>
      </Dialog>
    );

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    await waitFor(() => expect(screen.getByTestId('Dialog').children[1]).toHaveClass('content'));
    await waitFor(() => expect(screen.getByText('Test content')).toBeVisible());
    await waitFor(() => expect(screen.getByText('Test content')).toHaveClass('content'));
  });

  test('should render correctly with title prop', async () => {
    render(
      <Dialog visible>
        <Dialog.Header title="Test title" />
      </Dialog>
    );

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    await waitFor(() => expect(screen.getByTestId('Dialog').children[1]).toHaveClass('header'));
    await waitFor(() => expect(screen.getByText('Test title')).toBeVisible());
    await waitFor(() => expect(screen.getByText('Test title')).toHaveClass('title'));
  });

  test('should render correctly with subtitle prop', async () => {
    render(
      <Dialog visible>
        <Dialog.Header title="Test title" subtitle="Test subtitle" />
      </Dialog>
    );

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    await waitFor(() => expect(screen.getByText('Test subtitle')).toBeVisible());
    await waitFor(() => expect(screen.getByText('Test subtitle')).toHaveClass('subtitle'));
  });

  test('should render correctly with icon prop', async () => {
    render(
      <Dialog visible>
        <Dialog.Header icon={<Icon icon={<Menu />} />} title="Test title" subtitle="Test subtitle" />
      </Dialog>
    );

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    await waitFor(() => expect(screen.getByTestId('Menu')).toBeVisible());
  });

  test('should render correctly with fullscreen prop', async () => {
    render(<Dialog fullscreen visible />);

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    await waitFor(() => expect(screen.getByTestId('Dialog')).toHaveClass('fullscreen'));
  });

  test('should render correctly with footer prop', async () => {
    render(
      <Dialog visible>
        <Dialog.Footer>
          <Button>OK</Button>
        </Dialog.Footer>
      </Dialog>
    );

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    await waitFor(() => expect(screen.getByRole('button')).toBeVisible());
    await waitFor(() => expect(screen.getByTestId('Dialog').lastChild).toHaveClass('footer'));
  });

  test('should render correctly with closeable prop', () => {
    render(<Dialog visible closable={false} />);

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    expect(screen.queryByTestId('Cross')).not.toBeInTheDocument();
  });

  test('should execute callBack on close button click', () => {
    const onCloseCallback = jest.fn();
    render(<Dialog visible onClose={onCloseCallback} />);

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    expect(onCloseCallback).toBeCalledTimes(0);
    userEvent.click(screen.getByTestId('Cross'));
    expect(onCloseCallback).toBeCalledTimes(1);
  });

  test('should execute callBack on click outside of dialog', async () => {
    const onCloseCallback = jest.fn();
    render(<Dialog visible onClose={onCloseCallback} />);

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    expect(onCloseCallback).toBeCalledTimes(0);
    userEvent.click(document.querySelector('.backdrop'));
    expect(onCloseCallback).toBeCalledTimes(1);
  });

  test('should execute callBack on escape button click', async () => {
    const onCloseCallback = jest.fn();
    render(<Dialog visible onClose={onCloseCallback} />);

    expect(screen.getByTestId('Dialog')).toMatchSnapshot();
    expect(onCloseCallback).toBeCalledTimes(0);
    userEvent.keyboard('{esc}');
    expect(onCloseCallback).toBeCalledTimes(1);
  });

  test('should execute onClose callBack once on escape button click', async () => {
    const onCloseCallback = jest.fn();
    const onCloseCallback2 = jest.fn();
    const onCloseCallback3 = jest.fn();
    const { rerender } = render(
      <Dialog visible onClose={onCloseCallback}>
        <Dialog.Content>
          <Dialog visible onClose={onCloseCallback2}>
            <Dialog.Content>
              <Dialog visible onClose={onCloseCallback3}>
                <Dialog.Content>Dialog</Dialog.Content>
              </Dialog>
            </Dialog.Content>
          </Dialog>
        </Dialog.Content>
      </Dialog>
    );

    expect(screen.queryAllByTestId('Dialog')[0]).toMatchSnapshot();
    expect(onCloseCallback).toBeCalledTimes(0);
    expect(onCloseCallback2).toBeCalledTimes(0);
    expect(onCloseCallback3).toBeCalledTimes(0);
    userEvent.keyboard('{esc}');
    expect(onCloseCallback).toBeCalledTimes(0);
    expect(onCloseCallback2).toBeCalledTimes(0);
    expect(onCloseCallback3).toBeCalledTimes(1);

    onCloseCallback.mockClear();
    onCloseCallback2.mockClear();
    onCloseCallback3.mockClear();
    rerender(
      <Dialog visible onClose={onCloseCallback}>
        <Dialog.Content>
          <Dialog visible onClose={onCloseCallback2}>
            <Dialog.Content>
              <Dialog onClose={onCloseCallback3}>
                <Dialog.Content>Dialog</Dialog.Content>
              </Dialog>
            </Dialog.Content>
          </Dialog>
        </Dialog.Content>
      </Dialog>
    );
    userEvent.keyboard('{esc}');
    expect(onCloseCallback).toBeCalledTimes(0);
    expect(onCloseCallback2).toBeCalledTimes(1);
    expect(onCloseCallback3).toBeCalledTimes(0);

    onCloseCallback.mockClear();
    onCloseCallback2.mockClear();
    onCloseCallback3.mockClear();
    rerender(
      <Dialog visible onClose={onCloseCallback}>
        <Dialog.Content>
          <Dialog onClose={onCloseCallback2}>
            <Dialog.Content>
              <Dialog onClose={onCloseCallback3}>
                <Dialog.Content>Dialog</Dialog.Content>
              </Dialog>
            </Dialog.Content>
          </Dialog>
        </Dialog.Content>
      </Dialog>
    );
    userEvent.keyboard('{esc}');
    expect(onCloseCallback).toBeCalledTimes(1);
    expect(onCloseCallback2).toBeCalledTimes(0);
    expect(onCloseCallback3).toBeCalledTimes(0);
  });

  test('should execute onClose callBack once on escape button click. Internal dialog closed by cross icon', async () => {
    const onCloseCallback = jest.fn();
    const onCloseCallback2 = jest.fn();
    const { rerender } = render(
      <Dialog visible onClose={onCloseCallback}>
        <Dialog.Content>
          <Dialog visible onClose={onCloseCallback2}>
            <Dialog.Content>content</Dialog.Content>
          </Dialog>
        </Dialog.Content>
      </Dialog>
    );

    expect(screen.queryAllByTestId('Dialog')[0]).toMatchSnapshot();
    expect(onCloseCallback).toBeCalledTimes(0);
    expect(onCloseCallback2).toBeCalledTimes(0);
    userEvent.click(screen.queryAllByTestId('Cross')[1]);
    expect(onCloseCallback).toBeCalledTimes(0);
    expect(onCloseCallback2).toBeCalledTimes(1);

    onCloseCallback.mockClear();
    onCloseCallback2.mockClear();
    rerender(
      <Dialog visible onClose={onCloseCallback}>
        <Dialog.Content>
          <Dialog onClose={onCloseCallback2}>
            <Dialog.Content>content</Dialog.Content>
          </Dialog>
        </Dialog.Content>
      </Dialog>
    );
    userEvent.keyboard('{esc}');
    expect(onCloseCallback).toBeCalledTimes(1);
    expect(onCloseCallback2).toBeCalledTimes(0);
  });
});
