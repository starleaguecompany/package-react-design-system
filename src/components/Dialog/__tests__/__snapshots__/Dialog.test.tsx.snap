// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should execute callBack on click outside of dialog 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</div>
`;

exports[`Dialog should execute callBack on close button click 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</div>
`;

exports[`Dialog should execute callBack on escape button click 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</div>
`;

exports[`Dialog should execute onClose callBack once on escape button click 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
  <div
    class="content"
  />
</div>
`;

exports[`Dialog should execute onClose callBack once on escape button click. Internal dialog closed by cross icon 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
  <div
    class="content"
  />
</div>
`;

exports[`Dialog should render correctly with attributes 1`] = `
<div
  class="test-class container"
  data-qa="Dialog"
  id="test-id"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</div>
`;

exports[`Dialog should render correctly with closeable prop 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
/>
`;

exports[`Dialog should render correctly with content 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
  <div
    class="content"
  >
    Test content
  </div>
</div>
`;

exports[`Dialog should render correctly with footer prop 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
  <div
    class="footer container direction-horizontal justify-stretch"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <button
      class="container size-44 color-gray-outlined"
      data-qa="Button"
    >
      <span
        class="text"
      >
        OK
      </span>
    </button>
  </div>
</div>
`;

exports[`Dialog should render correctly with fullscreen prop 1`] = `
<div
  class="container fullscreen"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(1.05) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</div>
`;

exports[`Dialog should render correctly with icon prop 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
  <div
    class="header container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <div
      class="icon"
    >
      <span
        class="container size-20 shape-round"
        data-qa="Icon"
      >
        <svg
          data-qa="Menu"
          fill="none"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 5h12M4 10h12M4 15h12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </span>
    </div>
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
      style="column-gap: 8px; row-gap: 8px;"
    >
      <div
        class="title"
      >
        Test title
      </div>
      <div
        class="subtitle"
      >
        Test subtitle
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog should render correctly with no props 1`] = `<div />`;

exports[`Dialog should render correctly with subtitle prop 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
  <div
    class="header container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
      style="column-gap: 8px; row-gap: 8px;"
    >
      <div
        class="title"
      >
        Test title
      </div>
      <div
        class="subtitle"
      >
        Test subtitle
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog should render correctly with title prop 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
  <div
    class="header container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
      style="column-gap: 8px; row-gap: 8px;"
    >
      <div
        class="title"
      >
        Test title
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog should render correctly with visible prop 1`] = `
<div
  class="container"
  data-qa="Dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  <span
    class="close container size-28 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</div>
`;
